addons/api/tests/unit/utils/indexed-db-query-test.js

```diff
@@ -494,4 +496,32 @@ module('Unit | Utility | indexed-db-query', function (hooks) {
 
     assert.strictEqual(result.length, 1);
   });
+
+  test('it filters on boolean type fields with truthy value', async function (assert) {
+    const authMethod = this.server.create('auth-method', {
+      name: 'Generated_auth_method',
+      scopeId: scope1.id,
+    });
+    scope1.update({ primaryAuthMethodId: authMethod.id });
+    await seedIndexDb('auth-method', store, indexedDb, this.server);
+    const query = { filters: { is_primary: [{ equals: 'true' }] } };
+
+    const result = await queryIndexedDb(indexedDb.db, 'auth-method', query);
+
+    assert.strictEqual(result.length, 1);
+  });
+
+  test('it filters on boolean type fields with falsy value', async function (assert) {
+    const authMethod = this.server.create('auth-method', {
+      name: 'Generated_auth_method',
+      scopeId: scope1.id,
+    });
+    scope1.update({ primaryAuthMethodId: authMethod.id });
```

Comment:
Does this actually set the `is_primary` for the auth method itself? I don't think we have anything in the config that would update the auth method model when we patch this

Reply:
yeah, not in the config, only in the mirage [auth-method serializer ](https://github.com/hashicorp/boundary-ui/blob/main/addons/api/mirage/serializers/auth-method.js#L20)